indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractBatchIndexTask.java

```diff
@@ -719,7 +720,7 @@ protected boolean waitForSegmentAvailability(
 
     try (
         SegmentHandoffNotifier notifier = toolbox.getSegmentHandoffNotifierFactory()
-                                                 .createSegmentHandoffNotifier(segmentsToWaitFor.get(0).getDataSource())
+                                                 .createSegmentHandoffNotifier(segmentsToWaitFor.get(0).getDataSource(), taskId)
```

Comment:
Doing this will make the diff smaller and help with the coverage too.

```suggestion
                                                 .createSegmentHandoffNotifier(segmentsToWaitFor.get(0).getDataSource(), getId())
```

Reply:
Thanks for the review @kfaraz ! patched my PR with your suggestions